File: 121. Best_Time_to_Buy_and_Sell_Stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        vector<pair<int,int>> price_vec;
        
        int min_price = INT_MAX;
        int max_price = 0;
        
        for(int i = 0 ; i < prices.size() ; i++){
            min_price = min(min_price, prices[i]);
            max_price = max(max_price, prices[i] - min_price);
        }
            
            
        return max_price;
        
    }
};

File: 121. Best_Time_to_Buy_and_Sell_Stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int min_price = Integer.MAX_VALUE;
        int max_price = 0;
        
        for(int i = 0 ; i < prices.length ; i++){
            min_price = Math.min(min_price, prices[i]);
            max_price = Math.max(max_price, prices[i] - min_price);
        }
            
        return max_price;
        
        
    }
}